### ISSUE TITLE
multi_scan fails on macOS 10.14.1 with "NoMethodError: [!] undefined method `sub' for nil:NilClass"

### ISSUE THREAD

**Issue by jcharr1** [lyndsey-ferguson/fastlane-plugin-test_center#77]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

### Issue Description

After updating to macOS 10.14.1 and the update to the Mojave command line tools, multi_scan is suddenly failing with:

bundler: failed to load command: fastlane (/usr/local/opt/gems/bin/fastlane)
NoMethodError: [!] undefined method `sub' for nil:NilClass

This was working fine before on 10.14. I have another 2011 Mac that's too old for Mojave that's running High Sierra, and it's not having this problem. It's only on my Mojave Macs that I updated to 10.14.1. Running the standard fastlane scan on the same project seems to work ok. 

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>INFO [2018-10-31 11:29:07.67]: ------------------------------
INFO [2018-10-31 11:29:07.67]: --- Collecting system logs ---
INFO [2018-10-31 11:29:07.67]: ------------------------------
INFO [2018-10-31 11:29:07.74]: ▸ 10.14.1
INFO [2018-10-31 11:29:07.75]: $ xcrun simctl spawn AB755EB3-89FD-492A-A743-989F764406DF log collect --output [REDACTED]/test_output/system_logs-iPhone\ X_iOS_12.1.logarchive 2>/dev/null

ERROR [2018-10-31 11:29:08.31]: Exit status: 74
INFO [2018-10-31 11:29:08.67]: -------------------
INFO [2018-10-31 11:29:08.67]: --- Step: is_ci ---
INFO [2018-10-31 11:29:08.67]: -------------------
WARN [2018-10-31 11:29:08.67]: Lane Context:
INFO [2018-10-31 11:29:08.67]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios iphone-tests", :SIGH_PROFILE_TYPE=>"development", :MATCH_PROVISIONING_PROFILE_MAPPING=>{"[REDACTED]"=>"match Development [REDACTED]"}, :SCAN_DERIVED_DATA_PATH=>"/Volumes/DATA/Xcode/DerivedData/Innovative-gxkfuqpbxvqnykfqidprfwfphfuo", :SCAN_GENERATED_PLIST_FILES=>[], :SCAN_GENERATED_PLIST_FILE=>nil}
INFO [2018-10-31 11:29:08.67]: Successfully generated documentation at path '[REDACTED]/fastlane/README.md'

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | hostname                   | 0           |
| 2    | Verifying fastlane version | 0           |
| 3    | default_platform           | 0           |
| 4    | import_from_git            | 1           |
| 5    | is_ci                      | 0           |
| 6    | hostname                   | 0           |
| 7    | cocoapods                  | 20          |
| 8    | is_ci                      | 0           |
| 9    | is_ci                      | 0           |
| 10   | match                      | 16          |
| 💥   | multi_scan                 | 40          |
| 12   | is_ci                      | 0           |
+------+----------------------------+-------------+

DEBUG [2018-10-31 11:29:08.68]: All plugins are up to date
ERROR [2018-10-31 11:29:08.68]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...
Search query: undefined method `sub' for nil

URL: https://api.github.com/search/issues?q=undefined%20method%20%60sub%27%20for%20nil+repo:fastlane/fastlane
➡️  Failure submitting screenshots to Apple with Deliver
    https://github.com/fastlane/fastlane/issues/13181 [open] 5 💬
    4 weeks ago

➡️  [Investigation] Make fastlane start faster
    https://github.com/fastlane/fastlane/issues/13440 [open] 27 💬
    57 minutes ago

➡️  `app_store_build_number` fails with `undefined method `current_build_number' for nil:NilClass (NoMethodError)`
    https://github.com/fastlane/fastlane/issues/10447 [closed] 12 💬
    18 Mar 2018

and 19 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60sub%27%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
DEBUG [2018-10-31 11:29:09.69]: All plugins are up to date
bundler: failed to load command: fastlane (/usr/local/opt/gems/bin/fastlane)
NoMethodError: [!] undefined method `sub' for nil:NilClass
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/actions/tests_from_xctestrun.rb:27:in `xctest_bundle_path'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/actions/tests_from_xctestrun.rb:15:in `block in xctestrun_tests'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/actions/tests_from_xctestrun.rb:14:in `each'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/actions/tests_from_xctestrun.rb:14:in `xctestrun_tests'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/actions/tests_from_xctestrun.rb:7:in `run'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/helper/test_collector.rb:58:in `testables_tests'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/helper/correcting_scan_helper.rb:54:in `scan_testable'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/helper/correcting_scan_helper.rb:41:in `block in scan'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/helper/correcting_scan_helper.rb:40:in `each'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/helper/correcting_scan_helper.rb:40:in `scan'
  /usr/local/opt/gems/gems/fastlane-plugin-test_center-3.5.8/lib/fastlane/plugin/test_center/actions/multi_scan.rb:23:in `run'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
  Fastfile:100:in `test_on_simulator'
  Fastfile:123:in `block (2 levels) in parsing_binding'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /usr/local/opt/gems/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /usr/local/opt/gems/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /usr/local/opt/gems/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /usr/local/opt/gems/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /usr/local/opt/gems/gems/fastlane-2.107.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
  /usr/local/opt/gems/gems/fastlane-2.107.0/bin/fastlane:23:in `<top (required)>'
  /usr/local/opt/gems/bin/fastlane:23:in `load'
  /usr/local/opt/gems/bin/fastlane:23:in `<top (required)>'
</pre>
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.1                                     |
| Ruby                        | 2.5.3                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.17.2 (Apple Git-113)          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.14.1 (18B75)                    |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.5.3/lib            |
| OpenSSL Version             | OpenSSL 1.0.2p  14 Aug 2018                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 10.1                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./Fastfile`</summary>

```ruby
# fastlane Fastfile - Contains lane definitions for batch fastlane actions.
#                     This is fastlane's main point of entry.
# Author: Jason Charrier
# Created: April 10, 2017

# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

my_hostname = sh "hostname"
ENV["SLACK_URL"] = [REDACTED]	
import_from_git(url: [REDACTED],
               path: 'fastlane/Fastfile-ios')

# define things unique to this project
def test_on_simulator(device_type, the_hostname)
	my_hostname = the_hostname
	
	# default to iPhone config
	devices = ["iPhone X"]
	ENV["SLACK_URL"] = [REDACTED]
	slack_url = "--slack_url [REDACTED]"
		
	if device_type == "iPad"
		ENV["SLACK_URL"] = [REDACTED]
		slack_url = "--slack_url [REDACTED]"		
		devices.clear()
		devices = ["iPad Pro (12.9-inch) (2nd generation)"]
	end
	
	# enable Slack notificaitons only when running under CI server
	if !is_ci?
		slack_url = ""
		ENV["SLACK_URL"] = ""
	end

	# callback function for multi_scan. Runs after each test attempt finishes.
	gather_code_coverage = lambda do |testrun_info|
		# collect our code coverage data only for the first run attempt as further
		# runs will only run failed tests
		try_attempt = testrun_info[:try_count]
		failed_tests = testrun_info[:failed]
		
		if failed_tests.size > 0 && is_ci?
			slack(
				message: "Failed tests: `#{failed_tests}`",
				default_payloads: [],
				success: false
			)
		end # // if
		
		if try_attempt == 1
			if is_ci?
				slack(
					message: "Gathering code coverage for test run attempt #{try_attempt} of 4",
					default_payloads: [],
					attachment_properties: { color: "#509BD2" }
				)
			end
			#xcov(scheme: "[REDACTED]")
			sh "xcov --scheme [REDACTED] --workspace ./[REDACTED].xcworkspace #{slack_url} --output_directory ./fastlane/xcov_report --only_project_targets true"
			slather(
				scheme: "[REDACTED]",
				workspace: "[REDACTED]/[REDACTED].xcworkspace",
				proj: "[REDACTED]/[REDACTED].xcodeproj",
				cobertura_xml: "true",
				jenkins: "true",
				output_directory: "[REDACTED]/slather_output",
				ignore: "[REDACTED]/ipadTest/openssl/*"
			)
		elsif is_ci?
			slack(
				message: "Finished test run attempt #{try_attempt} of 4",
				default_payloads: [],
				attachment_properties: { color: "#509BD2" }	
			)
		end # // if
	end # // gather_code_coverage

	if is_ci?
		slack(
			message: "#{device_type} Tests started on host: #{my_hostname}",
			#default_payloads: [],
			attachment_properties: { color: "#509BD2" }
		)
	end
	begin
		match(type: "development", readonly: true)
	rescue => ex
		UI.error("#{ex} - retry match and attempt to force recreate the profile")
		match(type: "development", force: true)
	end
	multi_scan(
		# uncomment the next line to only run a subset of tests
		#only_testing: ["[REDACTED]/DownloadManagerUITests"],
		fail_build: false, 
		try_count: 4,
		devices: devices,
		output_directory: "test_output",
		testrun_completed_block: gather_code_coverage,
		output_types: "junit"	
	)
	
	if is_ci?
		slack(
			message: "#{device_type} Tests finished on host: #{my_hostname}",
			default_payloads: []
		)
	end
end

platform :ios do	
	# Define lanes unique to this project
	desc "Runs ALL the iPhone tests"
	lane :"iphone-tests" do
		test_on_simulator "iPhone", my_hostname
	end # // iphonetests
	
	desc "Runs ALL the iPad tests"
	lane :"ipad-tests" do
		test_on_simulator "iPad", my_hostname
	end # // ipadtests	
	
	after_all do |lane|
		# This block is called, only if the executed lane was successful
		
		# cleanup
		
		# push changes from the develop branch to the test branches
		if git_branch == "develop"
			UI.message("Attempting to rebase test branches")
			sh "git checkout develop"
			sh "git stash && git stash clear"
			begin
				sh "git checkout ipad-tests"
			rescue => ex
				UI.error("#{ex}. Trying to checkout as new branch.")
				sh "git checkout -b ipad-tests origin/ipad-tests"
			end
			sh "git rebase develop"
			sh "git push origin ipad-tests"

			begin
				sh "git checkout iphone-tests"
			rescue => ex
				UI.error("#{ex}. Trying to checkout as new branch.")
				sh "git checkout -b iphone-tests origin/iphone-tests"
			end
			sh "git rebase develop"
			sh "git push origin iphone-tests"
			sh "git checkout develop"
			
			# Trigger the Jenkins job to rescan the branches now that we've
			# pushed the changes. 
			sh "curl -X POST [REDACTED]"
		end
	end # // after_all
	
end # // platform ios


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```
</details>

<details><summary>`./Appfile`</summary>

```ruby
# fastlane Appfile - Contains some basic app info for all fastlane actions
# Author: Jason Charrier
# Created: April 10, 2017

app_identifier "[REDACTED]" # The bundle identifier of your app
apple_id "[REDACTED]" # Your Apple email address

team_id "[REDACTED]" # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.107.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                      | Version | Update-Status |
| --------------------------- | ------- | ------------- |
| fastlane-plugin-test_center | 3.5.8   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                         | Version      |
| --------------------------- | ------------ |
| slack-notifier              | 2.3.2        |
| atomos                      | 0.1.3        |
| claide                      | 1.0.2        |
| colored2                    | 3.1.2        |
| nanaimo                     | 0.2.6        |
| xcodeproj                   | 1.7.0        |
| rouge                       | 2.0.7        |
| xcpretty                    | 0.3.0        |
| terminal-notifier           | 1.8.0        |
| unicode-display_width       | 1.4.0        |
| terminal-table              | 1.8.0        |
| plist                       | 3.4.0        |
| addressable                 | 2.5.2        |
| multipart-post              | 2.0.0        |
| word_wrap                   | 1.0.0        |
| public_suffix               | 2.0.5        |
| tty-screen                  | 0.6.5        |
| tty-cursor                  | 0.6.0        |
| tty-spinner                 | 0.8.0        |
| babosa                      | 1.0.2        |
| colored                     | 1.2          |
| highline                    | 1.7.10       |
| commander-fastlane          | 4.4.6        |
| excon                       | 0.62.0       |
| faraday                     | 0.15.3       |
| unf_ext                     | 0.0.7.5      |
| unf                         | 0.1.4        |
| domain_name                 | 0.5.20180417 |
| http-cookie                 | 1.0.3        |
| faraday-cookie_jar          | 0.0.6        |
| fastimage                   | 2.1.4        |
| gh_inspector                | 1.1.3        |
| json                        | 2.1.0        |
| mini_magick                 | 4.5.1        |
| multi_json                  | 1.13.1       |
| multi_xml                   | 0.6.0        |
| rubyzip                     | 1.2.2        |
| security                    | 0.1.3        |
| xcpretty-travis-formatter   | 1.0.0        |
| dotenv                      | 2.5.0        |
| faraday_middleware          | 0.12.2       |
| naturally                   | 2.2.0        |
| simctl                      | 1.6.5        |
| uber                        | 0.1.0        |
| declarative                 | 0.0.10       |
| declarative-option          | 0.1.0        |
| representable               | 3.0.4        |
| retriable                   | 3.1.2        |
| mime-types-data             | 3.2018.0812  |
| mime-types                  | 3.2.2        |
| jwt                         | 2.1.0        |
| signet                      | 0.11.0       |
| memoist                     | 0.16.0       |
| os                          | 1.0.0        |
| googleauth                  | 0.6.7        |
| httpclient                  | 2.8.3        |
| google-api-client           | 0.23.9       |
| emoji_regex                 | 0.1.1        |
| CFPropertyList              | 3.0.0        |
| bundler                     | 1.17.1       |
| openssl                     | 2.1.2        |
| date                        | 1.0.0        |
| fileutils                   | 1.1.0        |
| mini_portile2               | 2.3.0        |
| nokogiri                    | 1.8.5        |
| ipaddr                      | 1.2.0        |
| xctest_list                 | 1.1.6        |
| fastlane-plugin-test_center | 3.5.8        |
</details>

*generated on:* **2018-10-31**
</details></pre>
</details>

**Comment by lyndsey-ferguson** [comment]
Can you attach the `xctestrun` file? There is an issue in parsing the file.

**Comment by jcharr1** [comment]
Sure. Here you go. 
[Innovative_iphonesimulator12.1-x86_64.xctestrun.zip](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/files/2539118/Innovative_iphonesimulator12.1-x86_64.xctestrun.zip)

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.5.9](
https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.5.9)
